src/components/Shop/BasketWindow/index.tsx

```diff
@@ -1,17 +1,27 @@
-import React, { useContext, FC } from "react";
+import React, { useContext, FC, useState } from "react";
 import styled from "styled-components";
-import PurchaseButton from "./PurchaseButton";
 import BasketItem from "./BasketItem";
 import { OnlineOrange } from "utility/style";
 import { StateContext } from "state/state";
-
 import { calculateCartTotal } from "types/inventory";
+import Button from "../../../atoms/Button";
+import { setModalState } from "state/actions";
+import Modal from "components/Shop/Modal";
+import { modalTypes } from "types/modal";
 
 const BasketWindow: FC = () => {
-  const { state } = useContext(StateContext);
-  const { cart } = state;
+  const { state, dispatch } = useContext(StateContext);
+  const { cart, user, inventory, modalState } = state;
+
+  const totalPrice = calculateCartTotal(cart, inventory);
+  const insufficient = user!.balance - totalPrice <= 0 ? true : false;
 
-  const totalPrice = calculateCartTotal(cart, state.inventory);
+  const purchase = () => {
+    if (insufficient || totalPrice <= 0) return undefined;
+    else {
+      return dispatch(setModalState(modalTypes.PURCHASE));
+    }
```

Comment:
Because you use `return`, you can just remove the `else` and just return the statement, as this will in effect be the same.